vhdbuilder/packer/imagecustomizer/azlosguard/scripts/azlosguard-postinstall.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+
+set -e
+
+VHD_LOGS_FILEPATH=/opt/azure/vhd-install.complete
+
+required_env_vars=(
+    "IMG_SKU"
+    "CONTAINER_RUNTIME"
+    "SKU_NAME"
+)
+
+for v in "${required_env_vars[@]}"
+do
+    if [ -z "${!v}" ]; then
+        echo "$v was not set!"
+        exit 1
+    else
+        echo "$v is set to '${!v}'"
+    fi
+done
+
+FEATURE_FLAGS="${FEATURE_FLAGS:-}"
+
+echo "Starting build on $(date)" > ${VHD_LOGS_FILEPATH}
+
+source /opt/azure/containers/provision_source.sh
+
+# Fixup repart config from base image
+sed -i 's/Type=usr/Type=linux-generic/' /etc/repart.d/12-usr-a.conf
+rm /etc/repart.d/15-boot-b.conf  /etc/repart.d/16-usr-b.conf  /etc/repart.d/17-usr-hash-b.conf || true
+
+# Setup a symlink for lg-redirect-sysext
+mkdir -p /etc/extensions/lg-redirect-sysext/usr/local/
+mkdir -p /opt/bin
+ln -s /opt/bin /etc/extensions/lg-redirect-sysext/usr/local/bin
+# Bind mount /opt/bin to /usr/local/bin during the build since the redirect sysext is not running
+mount --bind /opt/bin /usr/local/bin
+trap "umount /usr/local/bin" EXIT
+
+# Link /opt/azure/containers to /home/packer for postinstall
+ln -s /opt/azure/containers /home/packer
+
+### pre-install-dependencies ###
+echo -e "\nnews.none                          -/var/log/messages" >> /etc/rsyslog.d/60-CIS.conf
+# Create dir for update_certs.path
+mkdir /opt/certs
+chmod 1755 /opt/certs
+# Use AKS Log Collector instead of WALA log collections
+echo -e "\n# Disable WALA log collection because AKS Log Collector is installed.\nLogs.Collect=n" >> /etc/waagent.conf
+
+### install-dependencies ###
+# Start containerd to allow container precaching
+containerd &
+CONTAINERD_PID=$!
+echo "Started containerd with PID $CONTAINERD_PID"
+# shellcheck disable=2064
+trap "kill $CONTAINERD_PID" EXIT
+
+# Precache packages and containers from components.json
+/opt/azure/containers/install-dependencies.sh
```

Comment:
so all the below is a direct replacement for post-install-dependencies.sh... this is making more sense now